docs/en/sql-reference/functions/string-search-functions.md

```diff
@@ -360,6 +360,97 @@ Extracts a fragment of a string using a regular expression. If ‘haystack’ do
 
 Extracts all the fragments of a string using a regular expression. If ‘haystack’ doesn’t match the ‘pattern’ regex, an empty string is returned. Returns an array of strings consisting of all matches to the regex. In general, the behavior is the same as the ‘extract’ function (it takes the first subpattern, or the entire expression if there isn’t a subpattern).
 
+## extractAllGroupsHorizontal {#extractallgroups-horizontal}
+
+Matches all groups of the `haystack` string using the `pattern` regular expression. 
```

Comment:
It's useful to specify the regular expression syntax. I suspect that these functions use the re2 syntax https://github.com/google/re2/wiki/Syntax, but I'm not sure